pyccel/codegen/printing/ccode.py

```diff
@@ -280,7 +280,7 @@ def _print_LiteralComplex(self, expr):
     def _print_PythonComplex(self, expr):
         self._additional_imports.add("complex")
         return self._print(PyccelAssociativeParenthesis(PyccelAdd(expr.real_part,
-                        PyccelMul(expr.imag_part, LiteralImaginaryUnit()))))
+                        PyccelMul(PythonFloat(expr.imag_part), LiteralImaginaryUnit()))))
```

Comment:
we don't need to add cast the real part to get the correct result, just the img part is enough.
maybe the problem will be also in Fortran lang, we just need to find the case.
I found this case in c when using numpy full, in the ndarray.c we are using there a Generic preprocessor to recognize the type of the literal number and here the problem happens.

Reply:
I suspect that this is because a `LiteralComplex` with a real part doesn't actually exist.
```
1+2j
```
Could be a `LiteralComplex`, but in fact it is read as:
```
PyccelAdd(LiteralInt(1),PyccelComplex(0,2))
```
One day we should remove the real part, but in the mean time in case it ever gets used, I think we should treat the real and imaginary parts in the same way

> maybe the problem will be also in Fortran lang, we just need to find the case.

If the problem can occur in fortran then it shouldn't be fixed in the printer

Reply:
For the real/imag part. It turns out a `LiteralComplex` with a real part does exist if you create it with a call to `complex`. Can you confirm that you don't have this problem with (or similar to fit the problem you found):
```
np.full(2,complex(5,6))
```

Reply:
for Fortran, it's already handled.
https://github.com/pyccel/pyccel/blob/master/pyccel/ast/builtins.py#L106L111

Reply:
This problem should be fixed by PR #572